### ISSUE TITLE
Draggable region in frameless window hides the resize handle on window border

### ISSUE THREAD

**Issue by whrool** [electron/electron#3022]
Firstly, i create a frameless window. Then i put a html element, such as a div which is draggable, into body element and the element occupied the whole body(margin is 0px), the resizable event can not be triggered when the mouse moves over the window frame. I don't know whether it is the case. I feel it is not convenient because i have to leave some margins between body and the div.

**Comment by zcbenz** [comment]
Can you provide an example app?

**Comment by whrool** [comment]
### index.html
```html
<!DOCTYPE html>
<html>
  <head>
    <title>Hello World!</title>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
     <style  type="text/css">
      * {
        margin: 0px;
        padding: 0px;
      }
      #caption {
        background-color: #aaffee;
        height: 100px;
        -webkit-app-region: drag;
      }
    </style>
  </head>
  <body>
    <div id="caption">
        Window Caption
    </div>
  </body>
</html>
```

### main.js
```javascript
var app = require('app');  // Module to control application life.
var BrowserWindow = require('browser-window');  // Module to create native browser window.
require('crash-reporter').start();
var mainWindow = null;
// Quit when all windows are closed.
app.on('window-all-closed', function() {
  if (process.platform != 'darwin')
    app.quit();
});

app.on('ready', function() {
  // Create the browser window.
  mainWindow = new BrowserWindow({width: 800,
                                  height: 600,
                                  "min-width": 800,
                                  "min-height": 600,
                                  "use-content-size": false,
                                  resizable: true,
                                  center: true,
                                  frame: false});

  // and load the index.html of the app.
  mainWindow.loadUrl('file://' + __dirname + '/index.html');
  // Emitted when the window is closed.
  mainWindow.on('closed', function() {
    mainWindow = null;
  });
});
```

### package.json
```json
{
  "name": "test",
  "productName": "Test",
  "main": "main.js",
  "version": "v0.0.0.1"
}
```

**Comment by zcbenz** [comment]
What did you mean by "resizable event"?

**Comment by whrool** [comment]
Sorry,i may make it obscure. I just call it by the BrowserWindow property of resizable. Actually, i mean it is HTTOP,HITTOPLEFT or such that. Whether should i cal it HTTOP,HTTOPLEFT or others ?

**Comment by zcbenz** [comment]
Maybe you can paste some screenshot? We need to know what is wrong and which behavior is expected, both your code and description didn't make that clear.

**Comment by whrool** [comment]
![image](https://cloud.githubusercontent.com/assets/7413974/10384134/2c908e40-6e6a-11e5-9761-f6422c6374ac.png)

**Comment by Algram** [comment]
I have added a short little video here that should demonstrate it.

https://dl.dropboxusercontent.com/u/17885899/bugelectron.mp4

**Comment by zcbenz** [comment]
After looking into this issue for more, I'm closing this as won't fix.

On Windows and Linux, it is very common for borders to not have resize handle, for example the border of the close button:

<img width="190" alt="screen shot 2016-06-07 at 4 38 56 pm" src="https://cloud.githubusercontent.com/assets/639601/15849596/57d675c4-2cce-11e6-862e-f4e5ef313df0.png">

If we force showing resize handle for all borders the behavior would actually not be expected.

So for users don't want to hide the resize handle, they should avoid marking the areas near the border as draggable.

**Comment by Lk2v** [comment]
I had the same problem just put a 

> -webkit-app-region: no-drag;

 on your div

> Firstly, i create a frameless window. Then i put a html element, such as a div which is draggable, into body element and the element occupied the whole body(margin is 0px), the resizable event can not be triggered when the mouse moves over the window frame. I don't know whether it is the case. I feel it is not convenient because i have to leave some margins between body and the div.

**Comment by OmkarRajam** [comment]
There is a workaround which you can use to make it work. See following link for detailed info:
https://stackoverflow.com/a/58431903/11167389

The trick is to not apply `-webkit-app-region: drag;` on header but rather applying it on a absolutely positioned div inside it. This div will have dimensions slightly shorter compared to header.  The gap that we left at borders is then used for resizing. We will keep its `z-index: -1` so that it won't appear above control buttons. We will also have to set `position:relative;` to header to make it work. 

In short, create a class like following 
```
.titlebar-drag-region {  /*added*/
  top: 4px;
  left: 4px;
  display: block;
  position: absolute;
  width: calc(100% - 8px);
  height: calc(100% - 8px);
  z-index: -1;
  -webkit-app-region: drag;
}
```
Then insert a div like `<div class="titlebar-drag-region"></div>` inside header.
